subsys/usb/class/cdc_acm.c

```diff
@@ -1048,7 +1048,7 @@ static const struct uart_driver_api cdc_acm_driver_api = {
 			    CONFIG_USB_CDC_ACM_DEVICE_NAME "_" #x,	\
 			    &cdc_acm_init, &cdc_acm_dev_data_##x,	\
 			    &cdc_acm_config_##x,			\
-			    APPLICATION,				\
+			    POST_KERNEL,				\
```

Comment:
You need to make sure it is initialized after USB stack init functions.

https://github.com/zephyrproject-rtos/zephyr/blob/46330f13ed6cd39621d34d54fd75b0023d2c18d4/subsys/usb/usb_device.c#L1429

Reply:
Which priority should be used then?

Reply:
So it should be started `earlier to be ready in time for OpenThread`, what is the priority of `OpenThread` process init?

Reply:
POST_KERNEL should work, hid-core and netusb are already POST_KERNEL.

Reply:
It may work ;)

Reply:
> So it should be started `earlier to be ready in time for OpenThread`, what is the priority of `OpenThread` process init?

OpenThread is started by ieee802154_init() which in turn is called by the individual radio drivers.

E.g. RF2xx starts with CONFIG_IEEE802154_RF2XX_INIT_PRIO: https://github.com/zephyrproject-rtos/zephyr/blob/86d1b7d2c8a5b0012793686ec324f145d38e9d9b/drivers/ieee802154/ieee802154_rf2xx.c#L877-L878

Reply:
So they are both `POST_KERNEL` and
https://github.com/zephyrproject-rtos/zephyr/blob/86d1b7d2c8a5b0012793686ec324f145d38e9d9b/drivers/ieee802154/Kconfig.rf2xx#L29-L37

and CDC ACM has

https://github.com/zephyrproject-rtos/zephyr/blob/86d1b7d2c8a5b0012793686ec324f145d38e9d9b/kernel/Kconfig#L598-L604

Maybe we can change it to 
https://github.com/zephyrproject-rtos/zephyr/blob/86d1b7d2c8a5b0012793686ec324f145d38e9d9b/kernel/Kconfig#L606-L611

Reply:
All current 802.15.4 drivers have a default priority of 80-81.